main.go

```diff
@@ -140,6 +152,13 @@ func main() {
 					}
 
 					if *event.Type == k8s.EventAdded || *event.Type == k8s.EventModified {
+						// run process until shutdown is requested via SIGTERM and SIGINT
```

Comment:
Shouldn't this be moved outside the `if` statement so regardless of event type we effectively stop watching?